part2/test-queue.cpp

```diff
@@ -1,3 +1,5 @@
+#include <assert.h>
+#include "str.h"
```

Comment:
`str.h` should not be included here, since we are allowed to use our own `string`/`object` implementations. You should include your String header in `queue.h`, which should resolve your problem. Please update your pull request.

Reply:
I don't believe this is how importing should work. The API specified by `queue.h` has no dependency on strings, so `str.h` should not be included in `queue.h`. The test cases in `test-queue.cpp` *do* explicitly depend on strings, and therefore `str.h` should be included in `test-queue.cpp`. Including `str.h` does not actually tie you down to a particular string API, it juts ties you down to the file.

I suspect the real issue with this import is that this repo's string API is titled `str.h`, as opposed to the name which I think people more commonly used: `string.h`.

A real solution to this problem would be renaming this repository's `str.h` to `string.h`.

If the owner of the repo would not be open to this, you can also use the following workaround: create a dummy `str.h` file, with the following contents:
```cpp
#pragma once
// include the real string api
#include "string.h"
```

This workaround will satisfy their test cases, but also let you continue to store your actual string API in `string.h`.